From 2a1359b98a323eec54360b5011507b093ff282cb Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 8 Nov 2022 00:36:28 +0100
Subject: [PATCH] [openconnect] Fix crash in certificate error dialog

We create a dialog, put a widget in it, then exec that dialog, auto-delete it, then crash when deleting that widget

Instead of manually deleting the widget parent it to the dialog so that it gets deleted automatically

BUG: 461365
---
 vpn/openconnect/openconnectauth.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/vpn/openconnect/openconnectauth.cpp b/vpn/openconnect/openconnectauth.cpp
index f7efe8e5..55057086 100644
--- a/vpn/openconnect/openconnectauth.cpp
+++ b/vpn/openconnect/openconnectauth.cpp
@@ -615,7 +615,11 @@ void OpenconnectAuthWidget::validatePeerCert(const QString &fingerprint, const Q
 #endif
 
     if (openconnect_check_peer_cert_hash(d->vpninfo, value.toUtf8().data())) {
-        auto widget = new QWidget();
+        QPointer<QDialog> dialog = new QDialog(this);
+        dialog->setAttribute(Qt::WA_DeleteOnClose);
+        dialog.data()->setWindowModality(Qt::WindowModal);
+
+        auto widget = new QWidget(dialog.data());
         QVBoxLayout *verticalLayout;
         QHBoxLayout *horizontalLayout;
         QLabel *icon;
@@ -658,9 +662,6 @@ void OpenconnectAuthWidget::validatePeerCert(const QString &fingerprint, const Q
         infoText->setWordWrap(true);
         certificate->setText(peerCert);
 
-        QPointer<QDialog> dialog = new QDialog(this);
-        dialog->setAttribute(Qt::WA_DeleteOnClose);
-        dialog.data()->setWindowModality(Qt::WindowModal);
         dialog->setLayout(new QVBoxLayout);
         auto buttons = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel, dialog);
         connect(buttons, &QDialogButtonBox::accepted, dialog.data(), &QDialog::accept);
@@ -676,7 +677,6 @@ void OpenconnectAuthWidget::validatePeerCert(const QString &fingerprint, const Q
         } else {
             *accepted = false;
         }
-        widget->deleteLater();
     } else {
         *accepted = true;
     }
-- 
GitLab

